src/language/templates/pybind/pyvisitor.cpp

```diff
@@ -103,25 +103,25 @@ namespace py = pybind11;
 
 {% for node in nodes %}
 void PyVisitor::visit_{{ node.class_name|snake_case }}(ast::{{ node.class_name }}& node) {
-    PYBIND11_OVERLOAD_PURE(void, Visitor, visit_{{ node.class_name|snake_case }}, node);
+    PYBIND11_OVERLOAD_PURE(void, Visitor, visit_{{ node.class_name|snake_case }}, std::ref(node));
```

Comment:
Someone needs to know all the secrete / hidden documentation of pybind11 🙄 🎉

Reply:
hidden or non-existing... this was after a lot of digging :/

Reply:
Ok...Worth an issue or PR in the upstream documentation then!

Reply:
oh wow.. i did not even know `std::ref` . amazing!

Reply:
+1 to Omar's comment